tests/wsl/install/ms_win_installation.pm

```diff
@@ -27,17 +27,27 @@ sub run {
     assert_screen 'windows-setup', 1000;
     send_key 'alt-n';      # next
     save_screenshot;
-    send_key 'alt-i';                                                           # install Now
+    send_key 'alt-i';      # install Now
     save_screenshot;
-    send_key 'alt-n';                                                           # next
+    send_key 'alt-n';      # next
     assert_screen 'windows-activate';
-    assert_and_click 'windows-no-prod-key';
-    assert_screen 'windows-select-system';
-    send_key_until_needlematch('windows-10-pro', 'down');
-    send_key 'alt-n';                                                           # select OS (Win 10 Pro)
-    assert_screen 'windows-license';
+    if (my $key = get_var('_SECRET_WINDOWS_10_PRO_KEY')) {
+        type_string $key;
+        send_key 'ret';
+        if (check_screen("windows-wrong-key", 60)) {
+            die('The provided product key didn\'t work...');
+        }
```

Comment:
Please use a multi-tag `assert_screen` with `match_has_tag` instead of `check_screen` with non-zero timeout to prevent introducing any timing dependant behaviour, to save test execution time as well as state more explicitly from the testers point of view what are the expected alternatives. For example:

```perl
assert_screen([qw(yast2_console-finished yast2_missing_package)]);
if (match_has_tag('yast2_missing_package')) {
    send_key 'alt-o';  # confirm package installation
    assert_screen 'yast2_console-finished';
}
```

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

Reply:
done